test/jdk/sun/security/krb5/Krb5NameEquals.java

```diff
@@ -22,57 +22,133 @@
+                );
+                if (krb5ConfigCheck2.getExitValue() != 0) {
+                    isNative = false;
+                }
+            }
         }
-        MECH = temp;
-    }
 
-    public static void main(String[] argv) throws Exception {
-        GSSManager mgr = GSSManager.getInstance();
-
-        boolean result = true;
-        // Create GSSName and check their equals(), hashCode() impl
-        GSSName name1 = mgr.createName(NAME_STR1,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-        GSSName name2 = mgr.createName(NAME_STR2,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-        GSSName name3 = mgr.createName(NAME_STR1,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-
-        if (!name1.equals(name1) || !name1.equals(name3) ||
-            !name1.equals((Object) name1) ||
-            !name1.equals((Object) name3)) {
-            System.out.println("Error: should be the same name");
-            result = false;
-        } else if (name1.hashCode() != name3.hashCode()) {
-            System.out.println("Error: should have same hash");
-            result = false;
+        if (isNative) {
+
+            final var testCommand = new String[]{"-Dsun.security.jgss.native=true",
+                    Krb5NameEquals.Krb5NameEqualsTest.class.getName()
+            };
+
+            final var result = ProcessTools.executeTestJava(testCommand);
+
+            if (result.getExitValue() != 0) {
+                if (os.equals("Linux") &&
+                    (arch.equals("x86_64") || arch.equals("amd64"))) {
+                    final var javaPropCommand = new String[]{System.getProperty("java.home"),
+                            "-XshowSettings:properties",
+                            "-version 2"
+                    };
+
+                    final var javaPropCommandResult = ProcessTools.executeTestJava(javaPropCommand);
+
+                    final var installationIssue = Arrays.stream(javaPropCommandResult.getStdout().split("\n"))
+                            .anyMatch(line -> line.contains("sun.arch.data.model") && line.contains("32"));
+
+                    if (installationIssue) {
+                        throw new SkippedException("""
+                                Running 32-bit JDK on 64-bit Linux. Maybe only 64-bit library is installed.
+                                Please manually check if this is the case. Treated as PASSED now.""");
+                    }
+                } else {
+                    result.shouldHaveExitValue(0);
+                }
+            } else {
+                result.shouldHaveExitValue(0);
+            }
+        } else {
+
+            final var testCommand = new String[]{"-Djava.security.krb5.realm=R",
+                    "-Djava.security.krb5.kdc=127.0.0.1",
+                    Krb5NameEquals.Krb5NameEqualsTest.class.getName()
+            };
+
+            final var result = ProcessTools.executeTestJava(testCommand);
```

Comment:
There is no need to run this in a sub-process. Consider create an individual `@ron main/othervm` with the system properties is probably enough.